src/storage/src/manifest/storage.rs

```diff
@@ -477,4 +575,75 @@ mod tests {
         let mut it = log_store.scan(0, 11).await.unwrap();
         assert!(it.next_log().await.unwrap().is_none());
     }
+
+    #[tokio::test]
+    async fn test_compress_backward_compatible() {
```

Comment:
You could override the compress_type manually to simulate the old manifest store
```rust
// Creates the manifest store
let mut store = ManifestObjectStore::new("/", object_store.clone());
// Manually disable compression
store.compress_type = UNCOMPRESS;
// write data
store.save();
// save checkpoint without compression
store.save_checkpoint();

store.compress_type = DEFAULT_COMPRESS_TYPE;
// test checkpoint
store.load_checkpoint();
// test scan
store.scan()
```